sound/soc/sof/pm.c

```diff
@@ -433,7 +433,7 @@ int snd_sof_resume(struct device *dev)
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
 	int ret;
 
-	if (sdev->s0_suspend) {
+	if (sdev->d0i3_suspend) {
```

Comment:
No.
You could already be in D0ix when you suspend to S0.

Reply:
In today's code, no, we haven't implemented runtime D0ix yet, and I am not sure if we do need to support that. 

I talked with some customers here, they actually not caring about tens of mW saving when the whole system is consuming Watts.

Just putting Audio DSP in either D0(any streams opened) or D3(runtime PM) when the system 
is in S0 can simplify our logic here a lot.

@plbossart what's your opinion here?

Reply:
@keyonjie I told you maybe 10 times we need to have D0ix supported on Ubuntu platforms so that we can test without dependencies on S0ix. that makes it 11 times.

what I would really want at this point is a state machine, which describes how we transition. and then you implement that state machine.

Working with incremental improvements and not following the guidance will not help progress.

Reply:
> @keyonjie I told you maybe 10 times we need to have D0ix supported on Ubuntu platforms so that we can test without dependencies on S0ix. that makes it 11 times.

I can feel your angry fire out of my screen, sorry about that, anyway, I got the answer "c" to my question 1/3, let me keep that in mind and make change here scaleable for the future.

> 
> what I would really want at this point is a state machine, which describes how we transition. and then you implement that state machine.
> 
> Working with incremental improvements and not following the guidance will not help progress.

Reply:
@keyonjie I am not angry or trying to get personal, we discussed this last week during our team meeting and I don't see why we need to revisit the same points that were already agreed on. Anyway I see below that you got my points so looking forward to the next revision.